usb-mount.sh

```diff
@@ -85,6 +97,21 @@ do_unmount()
 	${log} "Unmounted ${DEVICE} from ${MOUNT_POINT}"
         /bin/rmdir "${MOUNT_POINT}"
         sed -i.bak "\@${MOUNT_POINT}@d" /var/log/usb-mount.track
+        #
+        #Remove samba share if exist
+        #
+        #Get name share
+	SMB_SHARE_NAME=$(echo ${MOUNT_POINT} | sed "s/\/media\///g")
+	if [ -f /etc/samba/${SMB_SHARE_NAME}.conf ]; then
+		#Remove definition file
+		/bin/rm /etc/samba/${SMB_SHARE_NAME}.conf
+		#Remove include from smb.conf
+		sed -i "s/include = \/etc\/samba\/${SMB_SHARE_NAME}.conf//g" /etc/samba/smb.conf
+		#Delete all trailing blank lines at end of file 
+		sed -i -e :a -e '/^\n*$/{$d;N;};/\n$/ba' /etc/samba/smb.conf
```

Comment:
This can be avoided if `sed` expression in the previous line is used with `//d` sort of approach instead of `s//g`: some thing like `sed -i "/include = \/etc\/samba \/${SMB_SHARE_NAME}.conf/d" /etc/samba/smb.conf`